javascripts/2048.js

```diff
@@ -1,32 +1,347 @@
 $(document).ready(function() {
   console.log('ready!');
+
   $('body').keydown(function(event){
     var arrow_keys = [37, 38, 39, 40];
     if(arrow_keys.indexOf(event.which) > -1) {
       var tile = $('.tile');
       moveTile(tile, event.which);
+      addTile();
       event.preventDefault();
     }
   })
+
+  $('h6').click(function(){
+    gameStart();
+  })
 })
 
 function moveTile(tile, direction) {
-  var new_tile_value = tile.attr("data-val") * 2;
-  tile.attr("data-val", new_tile_value);
-  tile.text(new_tile_value);
+  // if (tilesCollide(direction); == true) {
+    // var new_tile_value = tile.attr("data-val") * 2;
+    // tile.attr("data-val", new_tile_value);
+    // tile.text(new_tile_value); // then also moves to next available position
+  // } else {
+    // they do not merge, they slide into the next available position
+  // }
+  var c0 = $("div[data-col='c0']");
+  var col0 = $.makeArray(c0);
+  var c1 = $("div[data-col='c1']");
+  var col1 = $.makeArray(c1);
+  var c2 = $("div[data-col='c2']");
+  var col2 = $.makeArray(c2);
+  var c3 = $("div[data-col='c3']");
+  var col3 = $.makeArray(c3);
+
+  var tile = col0.concat(col1, col2, col3);
```

Comment:
Considering this looks like it's grabbing tiles per column then mashing them together into a final list of all tiles, it might be easier to do this::
```
   var tile = $.makeArray($('.tile'));
```